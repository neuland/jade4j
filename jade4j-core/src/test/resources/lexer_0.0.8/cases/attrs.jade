a(href='/contact') contact
a(href='/save').button save
a(foo, bar, baz)
a(foo='foo, bar, baz', bar=1)
a(foo='((foo))', bar= (1) ? 1 : 0 )
select
  option(value='foo', selected) Foo
  option(selected, value='bar') Bar
a(foo="class:")
input(pattern='\\S+')

a(href='/contact') contact
a(href='/save').button save
a(foo bar baz)
a(foo='foo, bar, baz' bar=1)
a(foo='((foo))' bar= (1) ? 1 : 0 )
select
  option(value='foo' selected) Foo
  option(selected value='bar') Bar
a(foo="class:")
input(pattern='\\S+')
foo(terse="true")

- var attrs = {foo: 'bar', bar: '<baz>'}

div&attributes(attrs)
