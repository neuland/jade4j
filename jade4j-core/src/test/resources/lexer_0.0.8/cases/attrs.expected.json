{"type":"tag","line":1,"val":"a","selfClosing":false}
{"type":"attrs","line":1,"attrs":[{"name":"href","val":"'/contact'","escaped":true}]}
{"type":"text","line":1,"val":"contact"}
{"type":"newline","line":2}
{"type":"tag","line":2,"val":"a","selfClosing":false}
{"type":"attrs","line":2,"attrs":[{"name":"href","val":"'/save'","escaped":true}]}
{"type":"class","line":2,"val":"button"}
{"type":"text","line":2,"val":"save"}
{"type":"newline","line":3}
{"type":"tag","line":3,"val":"a","selfClosing":false}
{"type":"attrs","line":3,"attrs":[{"name":"foo","val":true,"escaped":true},{"name":"bar","val":true,"escaped":false},{"name":"baz","val":true,"escaped":false}]}
{"type":"newline","line":4}
{"type":"tag","line":4,"val":"a","selfClosing":false}
{"type":"attrs","line":4,"attrs":[{"name":"foo","val":"'foo, bar, baz'","escaped":true},{"name":"bar","val":"1","escaped":true}]}
{"type":"newline","line":5}
{"type":"tag","line":5,"val":"a","selfClosing":false}
{"type":"attrs","line":5,"attrs":[{"name":"foo","val":"'((foo))'","escaped":true},{"name":"bar","val":"(1) ? 1 : 0","escaped":true}]}
{"type":"newline","line":6}
{"type":"tag","line":6,"val":"select","selfClosing":false}
{"type":"indent","line":7,"val":2}
{"type":"tag","line":7,"val":"option","selfClosing":false}
{"type":"attrs","line":7,"attrs":[{"name":"value","val":"'foo'","escaped":true},{"name":"selected","val":true,"escaped":false}]}
{"type":"text","line":7,"val":"Foo"}
{"type":"newline","line":8}
{"type":"tag","line":8,"val":"option","selfClosing":false}
{"type":"attrs","line":8,"attrs":[{"name":"selected","val":true,"escaped":true},{"name":"value","val":"'bar'","escaped":true}]}
{"type":"text","line":8,"val":"Bar"}
{"type":"outdent","line":9}
{"type":"tag","line":9,"val":"a","selfClosing":false}
{"type":"attrs","line":9,"attrs":[{"name":"foo","val":"\"class:\"","escaped":true}]}
{"type":"newline","line":10}
{"type":"tag","line":10,"val":"input","selfClosing":false}
{"type":"attrs","line":10,"attrs":[{"name":"pattern","val":"'\\\\S+'","escaped":true}]}
{"type":"newline","line":12}
{"type":"tag","line":12,"val":"a","selfClosing":false}
{"type":"attrs","line":12,"attrs":[{"name":"href","val":"'/contact'","escaped":true}]}
{"type":"text","line":12,"val":"contact"}
{"type":"newline","line":13}
{"type":"tag","line":13,"val":"a","selfClosing":false}
{"type":"attrs","line":13,"attrs":[{"name":"href","val":"'/save'","escaped":true}]}
{"type":"class","line":13,"val":"button"}
{"type":"text","line":13,"val":"save"}
{"type":"newline","line":14}
{"type":"tag","line":14,"val":"a","selfClosing":false}
{"type":"attrs","line":14,"attrs":[{"name":"foo","val":true,"escaped":true},{"name":"bar","val":true,"escaped":false},{"name":"baz","val":true,"escaped":false}]}
{"type":"newline","line":15}
{"type":"tag","line":15,"val":"a","selfClosing":false}
{"type":"attrs","line":15,"attrs":[{"name":"foo","val":"'foo, bar, baz'","escaped":true},{"name":"bar","val":"1","escaped":true}]}
{"type":"newline","line":16}
{"type":"tag","line":16,"val":"a","selfClosing":false}
{"type":"attrs","line":16,"attrs":[{"name":"foo","val":"'((foo))'","escaped":true},{"name":"bar","val":"(1) ? 1 : 0","escaped":true}]}
{"type":"newline","line":17}
{"type":"tag","line":17,"val":"select","selfClosing":false}
{"type":"indent","line":18,"val":2}
{"type":"tag","line":18,"val":"option","selfClosing":false}
{"type":"attrs","line":18,"attrs":[{"name":"value","val":"'foo'","escaped":true},{"name":"selected","val":true,"escaped":false}]}
{"type":"text","line":18,"val":"Foo"}
{"type":"newline","line":19}
{"type":"tag","line":19,"val":"option","selfClosing":false}
{"type":"attrs","line":19,"attrs":[{"name":"selected","val":true,"escaped":true},{"name":"value","val":"'bar'","escaped":true}]}
{"type":"text","line":19,"val":"Bar"}
{"type":"outdent","line":20}
{"type":"tag","line":20,"val":"a","selfClosing":false}
{"type":"attrs","line":20,"attrs":[{"name":"foo","val":"\"class:\"","escaped":true}]}
{"type":"newline","line":21}
{"type":"tag","line":21,"val":"input","selfClosing":false}
{"type":"attrs","line":21,"attrs":[{"name":"pattern","val":"'\\\\S+'","escaped":true}]}
{"type":"newline","line":22}
{"type":"tag","line":22,"val":"foo","selfClosing":false}
{"type":"attrs","line":22,"attrs":[{"name":"terse","val":"\"true\"","escaped":true}]}
{"type":"newline","line":24}
{"type":"code","line":24,"val":"var attrs = {foo: 'bar', bar: '<baz>'}","escape":false,"buffer":false}
{"type":"newline","line":26}
{"type":"tag","line":26,"val":"div","selfClosing":false}
{"type":"&attributes","line":26,"val":"attrs"}
{"type":"newline","line":27}
{"type":"eos","line":27}